src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.stategy.FruitOperationHandler;
+import core.basesyntax.stategy.OperationStrategy;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void processTransaction(List<FruitTransaction> fruitTransactions,
+                                   Map<String, Integer> inventory) {
+        for (FruitTransaction fruitTransaction: fruitTransactions) {
+            FruitOperationHandler handler =
+                    operationStrategy.getHandler(fruitTransaction.getOperation());
+            if (handler != null) {
+                handler.executeOperation(fruitTransaction, inventory);
+            } else {
+                throw new IllegalArgumentException("No handler for"
+                        + fruitTransaction.getOperation());
```

Comment:
Consider logging the error or providing more context in the error message when no handler is found for an operation. This can help with debugging and understanding the issue.